packages/rum/src/domain/record/serialize.ts

```diff
@@ -34,25 +35,33 @@ type ParentNodePrivacyLevel =
   | typeof NodePrivacyLevel.MASK
   | typeof NodePrivacyLevel.MASK_USER_INPUT
 
-export type SerializationContext = 'full-snapshot' | 'mutation'
+export enum SerializationContext {
+  INITIAL_FULL_SNAPSHOT,
+  SUBSEQUENT_FULL_SNAPSHOT,
+  MUTATION,
+}
 
 export interface SerializeOptions {
   document: Document
   serializedNodeIds?: Set<number>
   ignoreWhiteSpace?: boolean
   parentNodePrivacyLevel: ParentNodePrivacyLevel
   serializationContext: SerializationContext
+  elementsScrollPositions?: ElementsScrollPositions
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏it would be better to have this always defined. I know it is not used during `serializationContext === mutation`, but it is not obvious that those two options are linked together.

<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏If you want to avoid making this mandatory when it is not used, what about using a more explicit `SerializationContext` like this:

```ts
enum SerializationContextStatus {
  MUTATION,
  INITIAL_FULL_SNAPSHOT,
  SUBSEQUENT_FULL_SNAPSHOT,
}
type SerializationContext =
  {
    status: SerializationContextStatus.MUTATION
  } |
  {
    status: SerializationContextStatus.INITIAL_FULL_SNAPSHOT,
    elementsScrollPositions: ElementsScrollPosition,
  } |
  {
    status: SerializationContextStatus.SUBSEQUENT_FULL_SNAPSHOT,
    elementsScrollPositions: ElementsScrollPosition,
  }
```